packages/core/http/core-http-router-server-internal/src/kibana_response/kibana_file_response.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { KibanaFileResponseOptions } from '@kbn/core-http-server';
+import { KibanaResponse } from './kibana_response';
+
+/**
+ * A response payload that represents a request for a specific file
+ * @internal
+ */
+export class KibanaFileResponse extends KibanaResponse<undefined> {
+  constructor(
+    public readonly path: string,
+    public readonly options: KibanaFileResponseOptions = {}
+  ) {
+    super(0, undefined, options);
+  }
```

Comment:
That's the part I kinda have the most concerns with. This looks a little too much like trying to push a square shape in circle hole ihmo. OTOH this remains an implementation details from the POV of the consumer, so maybe it's fine.